conans/client/graph/compute_pid.py

```diff
@@ -59,7 +59,7 @@ def run_validate_package_id(conanfile):
     # IMPORTANT: This validation code must run before calling info.package_id(), to mark "invalid"
     if hasattr(conanfile, "validate"):
         with conanfile_exception_formatter(conanfile, "validate"):
-            with conanfile_remove_attr(conanfile, ['cpp_info'], "validate"):
+            with conanfile_remove_attr(conanfile, ['cpp_info', 'settings', 'options'], "validate"):
```

Comment:
Missing a test for this?

Reply:
This is `validate` method, not `package_id`

Reply:
Added tests